File: src/programmers/level_1/자료구조/LV2_프로세스.py
# 10:06~10:47
from collections import deque


def solution(priorities, location):
    q = deque([(v, i) for i, v in enumerate(priorities)])
    s = sorted([(-v, v) for _, v in enumerate(priorities)])

    idx = 0
    while True:
        cur = q.popleft()
        if s[idx][1] == cur[0]:
            idx += 1
            if cur[1] == location: return idx
        else:
            q.append(cur)

File: src/programmers/level_1/자료구조/LV2_프로세스.kt
import kotlin.collections.ArrayDeque
class Solution {
    fun solution(priorities: IntArray, location: Int): Int {
        val q=ArrayDeque<List<Int>>()
        val priorityQ=ArrayDeque<Int>()
        
        priorities.forEachIndexed{i,v->
            priorityQ.add(v)
            q.add(listOf(i,v))
        }
        
        priorityQ.sort()
        var count=1
        while(q.isNotEmpty()){
            val current=q.removeFirst()
            if(current[1]==priorityQ.last()){
                priorityQ.removeLast()
                if(current[0]==location){
                    return count
                }
                count++
            }else{
                q.add(current)
            }
        }
        
        return count
    }
}